subsys/bluetooth/services/ots/Kconfig

```diff
@@ -69,6 +59,45 @@ config BT_OTS_OLCP_GO_TO_SUPPORT
 	bool "Support OLCP Go To Operation"
 	default y
 
+config BT_OTS_OBJ_NAME_WRITE_SUPPORT
+	bool "Support object name write"
+
+endif # BT_OTS
+
+#### Object Transfer Service Client ################################
+
+config BT_OTS_CLIENT
+	bool "Object Transfer Service Client [Experimental]"
+	select BT_L2CAP_DYNAMIC_CHANNEL
+	select BT_GATT_CLIENT
```

Comment:
> but I've seen equally many requests to avoid depends on as that hides the settings from the menuconfig GUI.

If settings are forming a hierachy, then `depends on` is usually the right choice.

Just think of it as:
```
config FOO
    bool

if FOO

config FOO_SUBFEATURE
    bool

endif
```

that is identical to.
```
config FOO
    bool

config FOO_SUBFEATURE
    bool
    depends on FOO
```

because `BT_GATT_CLIENT` and `BT_OTS_CLIENT` are on parallel paths in the tree, you don't see this as clear.

The risk of mixing, just because some don't like `that hides the settings from the menuconfig GUI.` is risky because it may easily lead to circular dependencies. 
As well as the `select` bears a higher risk of the stuck-symbol syndrome.

Reply:
@tejlmand Not related to this PR, but for e.g. Bluetooth LE Audio, we are probably looking at something like 10-20 dependencies just for that feature. In some ways doing something like `CONFIG_BT_AUDIO_HEARING_AID=y` instead of enabling those 10-20 dependencies is perhaps also more user friendly (although by hiding away a lot of what has been enabled). 

In such as case, would you be in favor of `depends on` or `select`? 

I'm OK with using `depends on` where possible instead of `select`, as that is more intuitive for me (but I also never use the GUI).

Reply:
@tejlmand What was the conclusion here?  Is this something I should change?
I guess that with "depends" the person doing the configuration would need to _know_ all the dependencies, in order to be able to turn on the feature?  That seems unlikely to be the case.

Reply:
>  but for e.g. Bluetooth LE Audio, we are probably looking at something like 10-20 dependencies just for that feature. In some ways doing something like CONFIG_BT_AUDIO_HEARING_AID=y instead of enabling those 10-20 dependencies is perhaps also more user friendly (although by hiding away a lot of what has been enabled).

I'm not against `selects` in general, they can make a lot of sense.
What I was asking about, was in cases where something looks like a configuration tree, like:
```
--- BT
      |--- GATT_CLIENT
            |--- BT_OTS_CLIENT
```

but because the tree in Kconfig is actually constructed as this:
```
--- BT
      |--- GATT_CLIENT
      |--- BT_OTS_CLIENT
```

it becomes less visible.
It could be i'm mistaking, and you really want gatt client and ots client on same level, but to me, in this particular case it more looks as OTS client is depending on GATT client.

For the audio, I don't believe those 10-20 dependencies are forming something that could look like a heriachy.
More settings on a common level, correct ?
Hope this answers your question @Thalley 

> @tejlmand What was the conclusion here?

I don't think there is a conclusion yet.
This was more an observation, and i'm trying to understand the reason.
Also, if this is a general way things have been done in bluetooth, then I don't believe a single PR should deviate from that.
But then it could be somethingto consider if structure is correct.

>  Is this something I should change?

I don't think so atm.
This is also why I just posted this as a comment, to try to understand this part better.